lib/wor/paginate/formatters/base.rb

```diff
@@ -0,0 +1,61 @@
+require 'wor/paginate/utils/uri_helper'
+
+module Wor
+  module Paginate
+    module Formatters
+      class Base
+        include Utils::UriHelper
+        attr_accessor :adapter, :content, :formatter, :options
+
+        def initialize(adapter, options = {})
+          @adapter = adapter
+          @options = options
+        end
+
+        def format # rubocop: disable Metrics/MethodLength
+          {
+            page: serialized_content,
+            count: count,
+            total_pages: total_pages,
+            total_count: total_count,
+            current_page: current_page,
+            previous_page: previous_page,
+            next_page: next_page,
+            next_page_url: page_url(next_page),
+            previous_page_url: page_url(previous_page)
+          }
+        end
+
+        protected
+
+        delegate :count, :total_count, :total_pages, :previous_page, :next_page, to: :adapter
+
+        def current_page
+          adapter.page.to_i
+        end
+
+        def paginated_content
+          @content ||= adapter.paginated_content
+        end
+
+        def serialized_content
+          paginated_content.as_json
```

Comment:
:thinking: every paginated_content respond to `as_json` method? I'm not really sure :sweat_smile:

Reply:
This is a generic formatter, which calls the `as_json` method to invoke the serializer used in the project if you verify in the other formatters this method `serialized_content` is overwritten to use specific methods of Panko and AMS